spec/spec_helper.rb

```diff
@@ -36,3 +36,62 @@
   # triggering implicit auto-inclusion in groups with matching metadata.
   config.shared_context_metadata_behavior = :apply_to_host_groups
 end
+
+def stub_fastly_get_service(service_id)
+  requests = []
+
+  # Fastly#get_service. Return a service with two VCL "versions" (https://docs.fastly.com/api/config#version)
+  requests << stub_request(:get, "https://api.fastly.com/service/#{service_id}")
+    .to_return(body: File.read("spec/fixtures/fastly-get-service-response.json"))
+
+  # We clone the latest active VCL version, which returns the latest version
+  requests << stub_request(:put, "https://api.fastly.com/service/#{service_id}/version/2/clone")
+    .to_return(body: File.read("spec/fixtures/fastly-put-clone.json"))
+end
+
+def stub_fastly_apply_service_changes(service_id, is_bouncer: false)
+  requests = stub_fastly_get_service(service_id)
+
+  # We first check the latest version
+  requests << stub_request(:get, "https://api.fastly.com/service/#{service_id}/version/3/vcl/main")
+  .to_return(body: "{}")
+
+  # We then delete it
+  requests << stub_request(:delete, "https://api.fastly.com/service/#{service_id}/version/3/vcl/main")
+    .to_return(body: "{}")
+
+  # Stub calls to delete the "UI objects"
+  service_type = %w[healthcheck cache_settings request_settings response_object header gzip]
+  service_type += %w[backend condition] if is_bouncer
+  service_type.each do |thing|
+    requests << stub_request(:get, "https://api.fastly.com/service/#{service_id}/version/3/#{thing}")
+      .to_return(body: "{}")
+  end
+
+  # We first check the latest version
+  requests << stub_request(:get, "https://api.fastly.com/service/#{service_id}/version/3/vcl/main")
+    .to_return(body: "{}")
+
+  # We then delete it
+  requests << stub_request(:delete, "https://api.fastly.com/service/#{service_id}/version/3/vcl/main")
+    .to_return(body: "{}")
```

Comment:
I'm confused as to why we're doing this again? (We've just done it on lines 55-61)